src/os.cc

```diff
@@ -92,11 +92,11 @@ int detail::utf16_to_utf8::convert(wstring_view s) {
 
   int length = WideCharToMultiByte(CP_UTF8, 0, s.data(), s_size, nullptr, 0,
                                    nullptr, nullptr);
-  if (length == 0) return GetLastError();
-  buffer_.resize(length + 1);
+  if (length == 0) return static_cast<int>(GetLastError());
```

Comment:
I suggest adding an inline function:

```
inline int get_windows_error() {
  static_cast<int>(GetLastError());
}
```
and using it to avoid casts in multiple places.